### PR TITLE
compile and integrate windows credential helper into installer

### PR DESCRIPTION
**Pull request by maxxnowak**
Gits source now contains an implementation for the windows credential
API. Lets teach git to remember usernames and passwords using this
helper.

Signed-off-by: Heiko Voigt <heiko.voigt@mahr.de>

### CODE REVIEW

**Comment by dscho** [comment]
@hvoigt what do you think?

@shiftkey would this interfere with GitHub for Windows?

@kusma do you think it makes sense (being the local credentials guru)?

**Comment by dscho** [comment]
@maxxnowak could you make sure that this Pull Request merges cleanly, please?

**Comment by kusma** [comment]
I don't understand, we did this work a long time ago... My installation of Git for Windows contains C:\Program Files (x86)\Git\libexec\git-core\git-credential-wincred.exe.

Further, we have this to build it:
https://github.com/msysgit/msysgit/blob/master/share/WinGit/release.sh#L84

And this should copy it to the installer:
https://github.com/msysgit/msysgit/blob/master/share/WinGit/copy-files.sh#L40

The only thing lacking is enabling it by default, which I'm skeptical to. If anything, this should be an install-time decision, not a global default.

In other words, I think things are OK as-is, no need for this patch. But perhaps the feature deserves better discoverability?

**Comment by shiftkey** [comment]
@dscho

> would this interfere with GitHub for Windows?

No issues with this change - we already tweak this config as part of our build process...

**Comment by kusma** [comment]
I think I'm just going to say: Nack.
